src/components/NcColorPicker/NcColorPicker.vue

```diff
@@ -386,6 +386,40 @@ export default {
 			this.$emit('input', color)
 
 		},
+
+		/**
+		 * Calculate luminance of provided hex color
+		 *
+		 * @param {string} color the hex color
+		 */
+		 calculateLuma(color) {
+			const [red, green, blue] = this.hexToRGB(color)
+			return (0.2126 * red + 0.7152 * green + 0.0722 * blue) / 255
+		},
+
+		/**
+		 * Convert hex color to RGB
+		 *
+		 * @param {string} hex the hex color
+		 */
+		 hexToRGB(hex) {
+			const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex)
+			return result
+				? [parseInt(result[1], 16), parseInt(result[2], 16), parseInt(result[3], 16)]
+				: null
+		},
+
+		/**
+		 * gets the contrast color to use for the user chosen color
+		 *
+		 * @return {string} representing the hex value to change the check color to contrast with the background color
+		 *
+		 */
+		 contrastColor() {
+			const black = '#000000'
+			const white = '#FFFFFF'
+			return (this.calculateLuma(this.currentColor) > 0.5) ? black : white
+		},
```

Comment:
This should be a computed property. It calculates a value that depends only on the component's state.

Using it as a method in the template makes the template less declarative and requires to check that this method is good in a performance aspect (it is called each time the component renders) and has no side effects.

Reply:
Agreed! Moving it to be a computed property, thanks!